libs/cimit-service/src/main/java/uk/gov/di/ipv/core/library/service/CiMitService.java

```diff
@@ -281,4 +379,70 @@ private void logLambdaExecutionError(
         message.values().removeIf(Objects::isNull);
         LOGGER.error(new StringMapMessage(message));
     }
+
+    private void logApiRequestError(CimitApiResponse failedResponse) {
+        LOGGER.error(LogHelper.buildErrorMessage(FAILED_API_REQUEST, failedResponse.reason()));
+    }
+
+    private void sendPostHttpRequest(
+            String endpoint, String govukSigninJourneyId, String ipAddress, String payload)
+            throws NonRetryableException, PostApiException, URISyntaxException, IOException {
+        var uri = getUriBuilderWithBaseApiUrl(endpoint).build();
+
+        var httpRequestBuilder = buildHttpRequest(uri, govukSigninJourneyId, ipAddress);
+        httpRequestBuilder.POST(HttpRequest.BodyPublishers.ofString(payload));
+
+        var response = sendHttpRequest(httpRequestBuilder.build());
+
+        var parsedResponse = OBJECT_MAPPER.readValue(response.body(), CimitApiResponse.class);
+
+        if (FAILED_RESPONSE.equals(parsedResponse.result())) {
+            logApiRequestError(parsedResponse);
+            throw new PostApiException(FAILED_API_REQUEST);
+        }
+    }
+
+    private HttpResponse<String> sendGetHttpRequest(
+            String userId, String govukSigninJourneyId, String ipAddress)
+            throws NonRetryableException, URISyntaxException, IOException {
+        var uriBuilder = getUriBuilderWithBaseApiUrl(GET_VCS_ENDPOINT);
+        var uri = uriBuilder.addParameter(USER_ID_PARAMETER, userId).build();
+
+        var httpRequestBuilder = buildHttpRequest(uri, govukSigninJourneyId, ipAddress).GET();
+
+        return sendHttpRequest(httpRequestBuilder.build());
+    }
+
+    private HttpRequest.Builder buildHttpRequest(
+            URI uri, String govukSigninJourneyId, String ipAddress) {
+        var requestBuilder =
+                HttpRequest.newBuilder()
+                        .uri(uri)
+                        .header(GOVUK_SIGNIN_JOURNEY_ID_HEADER, govukSigninJourneyId)
+                        .header(CONTENT_TYPE, ContentType.APPLICATION_JSON.toString())
+                        .header(X_API_KEY_HEADER, configService.getSecret(CIMIT_API_KEY));
+
+        if (ipAddress != null) {
+            requestBuilder.header(IP_ADDRESS_HEADER, ipAddress);
+        }
+        return requestBuilder;
+    }
+
+    private HttpResponse<String> sendHttpRequest(HttpRequest cimitHttpRequest)
+            throws NonRetryableException, IOException {
+        try {
+            LOGGER.info(LogHelper.buildLogMessage("Sending HTTP request to CiMit."));
+            return httpClient.send(cimitHttpRequest, HttpResponse.BodyHandlers.ofString());
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            throw new NonRetryableException(e);
```

Comment:
I don't think we should be using `Retryable`/`NonRetryableException` anywhere here